pkgs/by-name/lu/luau-lsp/package.nix

```diff
@@ -0,0 +1,50 @@
+{ fetchFromGitHub
+, stdenv
+, lib
+, cmake
+, gcc9
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "luau-lsp";
+  version = "1.28.1";
+
+  src = fetchFromGitHub {
+    owner = "JohnnyMorganz";
+    repo = "luau-lsp";
+    rev = finalAttrs.version;
+    hash = "sha256-5+m5tLX0DRHrV6yI/SXc39orpAu1DlBmpNQ/rt7LUjQ=";
+    fetchSubmodules = true;
+  };
+
+  # Build fails if this is false
+  dontUseCmakeConfigure = true;
+
+  nativeBuildInputs = [ cmake ]
+    ++ lib.optional stdenv.isLinux gcc9;
```

Comment:
Do we really need gcc9?

Reply:
Yes, the build fails to compile on version 13 so I used gcc9 because that is what is used on github actions of the [project](https://github.com/JohnnyMorganz/luau-lsp/actions/runs/8137576413/job/22236258736#step:3:7)

Reply:
What's the error? Has upstream maybe a fix for that already or it is a one line fix? Does upstream set Werror? I think it does. We want to patch that out to prevent unforeseen breakages in the future and reduce the chore on us https://github.com/JohnnyMorganz/luau-lsp/blob/131653d303ea89f183659d81e45babeb59f9f794/CMakeLists.txt#L76

Reply:
Sorry for the late reply.

When I use GCC version 13.2.0, I get the following error: https://gist.github.com/Eggflaw/d2682964b8733628fa35649af36d046f#file-build-log-L193-L255

I tried to temporarily remove the `Werror` option with substituteInPlace, the previous error pops up as a warning but another error pops up when it reaches to 100%
https://gist.github.com/Eggflaw/920dd55bf83471e7faeef289cbbf1b77#file-build-log2-L257-L272